### PR TITLE
Remove requests queue inside Host

### PR DESCRIPTION
**Pull request by dangleptr**
We don't need to hold all requests inside host.  There are two defects:
1.  Catch up data will be slow,  because we have to send request one by one.
2.  More memory  occupied due to the queue.

### CODE REVIEW

**Comment by dangleptr** [comment]
Jenkins go

File: src/kvstore/raftex/Host.cpp [hunk 1]
```diff
@@ -106,18 +107,13 @@ folly::Future<cpp2::AppendLogResponse> Host::appendLogs(
...
-                                    committedLogId,
-                                    lastLogTermSent,
-                                    lastLogIdSent)));
-                return requests_.back().first.getFuture();
+            if (cachingPromise_.size() <= FLAGS_max_outstanding_requests) {
+                pendingReq_ = std::make_tuple(term,
+                                              logId,
+                                              committedLogId,
+                                              lastLogTermSent,
+                                              lastLogIdSent);
```
Comment (wadeliuyi) [inline src/kvstore/raftex/Host.cpp]: I think the caller’s parameter name “currTerm, lastLogId, committedId, prevLogTerm, prevLogId” is more readable than here.

File: src/kvstore/raftex/Host.cpp [hunk 2]
```diff
@@ -185,8 +182,8 @@ folly::Future<cpp2::AppendLogResponse> Host::appendLogsInternal(
             {
                 std::lock_guard<std::mutex> g(self->lock_);
                 self->promise_.setValue(r);
+                self->cachingPromise_.setValue(r);
```
Comment (wadeliuyi) [inline src/kvstore/raftex/Host.cpp]: why the cached request no need retry?

File: src/kvstore/raftex/Host.cpp (hunk 1 above)
Comment (dangleptr) [reply src/kvstore/raftex/Host.cpp]: prevLogTerm, prevLogId sounds good

File: src/kvstore/raftex/Host.cpp (hunk 2 above)
Comment (dangleptr) [reply src/kvstore/raftex/Host.cpp]: Next request will fill the gap.  So we don't need retry here.

File: src/kvstore/raftex/Host.cpp [hunk 3]
```diff
@@ -138,20 +135,21 @@ folly::Future<cpp2::AppendLogResponse> Host::appendLogs(
...
         logIdToSend_ = logId;
-        lastLogTermSent_ = lastLogTermSent;
-        lastLogIdSent_ = lastLogIdSent;
+        lastLogTermSent_ = prevLogTerm;
+        lastLogIdSent_ = prevLogId;
         committedLogId_ = committedLogId;
 
-        promise_ = folly::Promise<cpp2::AppendLogResponse>();
+        promise_ = std::move(cachingPromise_);
+        cachingPromise_ = folly::SharedPromise<cpp2::AppendLogResponse>();
```
Comment (critical27) [inline src/kvstore/raftex/Host.cpp]: Do we need to reset pendingReq_ here?

Comment (dangleptr) [reply src/kvstore/raftex/Host.cpp]: Good catch!

**Pull request merged**